plugin/src/main/java/org/opensearch/ml/rest/RestMLStatsAction.java

```diff
@@ -124,33 +129,60 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
+                            && (finalMlStatsInput.retrieveAllClusterLevelStats()
+                                || finalMlStatsInput.getClusterLevelStats().contains(MLClusterLevelStat.ML_MODEL_COUNT))) {
+                            indexUtils
+                                .getNumberOfDocumentsInIndex(
+                                    ML_MODEL_INDEX,
+                                    QUERY_ALL_MODEL_META_DOC,
+                                    xContentRegistry,
+                                    ActionListener.wrap(modelCount -> {
+                                        clusterStatsMap.put(MLClusterLevelStat.ML_MODEL_COUNT, modelCount);
+                                        indexUtils.getNumberOfDocumentsInIndex(ML_CONNECTOR_INDEX, ActionListener.wrap(connectorCount -> {
+                                            clusterStatsMap.put(MLClusterLevelStat.ML_CONNECTOR_COUNT, connectorCount);
+                                            getNodeStats(finalMlStatsInput, clusterStatsMap, client, mlStatsNodesRequest, channel);
+                                        }, e -> {
+                                            String errorMessage = "Failed to get ML model count";
+                                            log.error(errorMessage, e);
+                                            onFailed(channel, RestStatus.INTERNAL_SERVER_ERROR, errorMessage, e);
+                                        }));
+                                    }, e -> {
+                                        String errorMessage = "Failed to get ML model count";
+                                        log.error(errorMessage, e);
+                                        onFailed(channel, RestStatus.INTERNAL_SERVER_ERROR, errorMessage, e);
+                                    })
+                                );
+                        } else {
+                            try {
                                 getNodeStats(finalMlStatsInput, clusterStatsMap, client, mlStatsNodesRequest, channel);
-                            }, e -> {
-                                String errorMessage = "Failed to get ML model count";
-                                log.error(errorMessage, e);
-                                onFailure(channel, RestStatus.INTERNAL_SERVER_ERROR, errorMessage, e);
-                            }));
-                        }, e -> {
-                            String errorMessage = "Failed to get ML model count";
-                            log.error(errorMessage, e);
-                            onFailure(channel, RestStatus.INTERNAL_SERVER_ERROR, errorMessage, e);
-                        })
-                    );
-            } else {
-                getNodeStats(finalMlStatsInput, clusterStatsMap, client, mlStatsNodesRequest, channel);
+                            } catch (IOException e) {
+                                throw new RuntimeException(e);
```

Comment:
Do we need to add this try/catch ? I don't see it in the old code

Reply:
editor was showing me: `Unhandled exception: java.io.IOException`